yarpl/include/yarpl/observable/Observable.h

```diff
@@ -167,6 +203,57 @@ Reference<Observable<T>> Observable<T>::subscribeOn(folly::Executor& executor) {
 }
 
 template <typename T>
+template <typename Function>
+Reference<Observable<T>> Observable<T>::doOnSubscribe(Function function) {
+  return details::createDoOperator(ref_from_this(this), std::move(function), [](const T&){}, [](auto){}, []{});
```

Comment:
it would not be possible, because the DoOperator is inferring the lambda type from the provided value. and since its an internal type, the user won't be facing this inconvenience.